### PR TITLE
Implement POST /api/articles/:id/comments

### PR DESCRIPTION
**Pull request by SamPetcher**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -153,3 +153,56 @@ describe('GET /api/articles/:article_id/comments', () => {
...
 })
 
+describe('POST /api/articles/:article_id/comments', () => {
+	// This end-point will take an object with the username and body props and insert in to the database. It should respond with the posted comment...
+	it('Should successfully post a comment and return it to the user', () => {
+	const commentPost = {username: 'butter_bridge', body: 'the vomit was white with black spots and a lady on our street owns a dalmation',}
+		return request(app)
+		.post('/api/articles/1/comments')
+		.send(commentPost)
+		.expect(200)
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: Posts will be a 201 Created status code, pls amend 👍🏾

File: __tests__/app.test.js [hunk 2]
```diff
@@ -153,3 +153,56 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+		.then( (comment) => {
+				expect(comment.body).toEqual(expect.objectContaining({
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: you should be sending data back on the comment key, e.g.
```
{ body: { comment: { comment_id: ...} } }
```

File: __tests__/app.test.js [hunk 3]
```diff
@@ -153,3 +153,56 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+		})
+	})
+	it('Should return an error when trying to post a non existent user', () => {
+	const commentPost = {username: 'Ken M', body: 'the vomit was white with black spots and a lady on our street owns a dalmation',}
+		return request(app)
+		.post('/api/articles/1/comments')
+		.send(commentPost)
+		.expect(404)
+		.then( (comment) => {
+			expect(comment.body.msg).toEqual("Invalid article")
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: This error message doesn't match what's wrong with the posted body (user being wrong). Would be worth amending this message to either be vaguer or reflect the fact it;s a malformed body

File: __tests__/app.test.js [hunk 4]
```diff
@@ -153,3 +153,56 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+		})
```
Comment (Poonam-raj) [inline __tests__/app.test.js:203]: More tests worth including for this endpoint:
- 201 when other properties are in the response body (ignores unnecessary properties)
- 400 when required fields are missing

File: controllers/errors.controllers.js [hunk 5]
```diff
@@ -8,6 +8,10 @@ if (err.code === '22P02') {
 	} 
 	else if (err.code === '22003') {
 		res.status(400).send({ msg:  "Invalid URL exceeds viable number of articles" })
+	}
+
+	else if (err.code === '23503') {
+		res.status(404).send({ msg:  "Invalid article" })
```
Comment (Poonam-raj) [inline controllers/errors.controllers.js]: I think this is the error that could reflect the fact it's actually invalid username possibly

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
Hey Sam, few more tests needed and then you're rocking and rolling with this PR - amend the code and throw it our way again

File: __tests__/app.test.js [hunk 6]
```diff
@@ -153,3 +153,82 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+        comment_id: expect.any(Number),
+        author: expect.any(String),
+        body: expect.any(String),
+        created_at: expect.any(String),
+        votes: expect.any(Number),
+      }))
+
+		})
+})
+	it('Should 404 when the user doesnt provide the correct field...', () => {
```
Comment (B7Bprog) [inline __tests__/app.test.js]: Please change this code to 400 in the test description.

**Review by B7Bprog — APPROVED** [review]
Hey Sam, please check my comment and fix that typo, then feel free to merge! 👍

**Pull request merged**